### PR TITLE
LO finetune fix

### PR DESCRIPTION
**Pull request by ik1xpv**
Fix the tune for LO feature.

### CODE REVIEW

**Comment by ik1xpv** [comment]
DRAFT to be discharged. I
mbintuce step 4 (250) kHz works. I expected step 2 125 kHz .
Many thing to be understood. :-)

File: ExtIO_sddc/r2iq.cpp [hunk 1]
```diff
@@ -72,11 +72,10 @@ int r2iqControlClass::Setdecimate(int dec)
...
 {
-	this->mtunebin = int(offset * halfFft / mratio[mdecimation] + 0.5);
-	float delta = offset - ((float)this->mtunebin * mratio[mdecimation] / halfFft);
-	float ret = -delta / mratio[mdecimation];
-	DbgPrintf("tunebin : %f -> %d delta %f (%f)\n", offset, this->mtunebin, delta, ret);
-
+	this->mtunebin = int(offset * halfFft/4  + 0.5 ) * 4;  // mtunebin step 4 bin  ? 
+	float delta = offset - ((float)this->mtunebin  / halfFft);
+	float ret = - delta * mratio[mdecimation]; // ret increases with higher decimation
+	DbgPrintf("LOfreq %d mtunebin %d delta %f (%f)\n", (int) (offset * ADC_FREQ/2), this->mtunebin, delta, ret);
```
Comment (howard0su) [inline ExtIO_sddc/r2iq.cpp:78]: can you print the LOfrq in RadioHandler? I don't want r2iq depends on ADC_FREQ. I am working on a change to have a dynamic ADC frequency.

File: ExtIO_sddc/r2iq.cpp [hunk 2]
```diff
@@ -385,11 +384,11 @@ void * r2iqControlClass::r2iqThreadf(r2iqThreadArg *th) {
 					}
 					else
 					{
-						th->inFreqTmp[m][0] = 0.0;
-						th->inFreqTmp[m][1] = 0.0;
+						th->inFreqTmp[m][0] = 0;
+						th->inFreqTmp[m][1] = 0;
```
Comment (howard0su) [inline ExtIO_sddc/r2iq.cpp:388]: good catch

File: ExtIO_sddc/r2iq.cpp [hunk 3]
```diff
@@ -398,16 +397,17 @@ void * r2iqControlClass::r2iqThreadf(r2iqThreadArg *th) {
 					if (mm >= 0) // corrects off limits
 					{
 						th->inFreqTmp[mfft / 2 + m][0] = (th->ADCinFreq[mm][0] * filter[fm][0] +
-							th->ADCinFreq[mm][1] * filter[fm][1]);
+							th->ADCinFreq[mm ][1] * filter[fm][1]);
```
Comment (howard0su) [inline ExtIO_sddc/r2iq.cpp:400]: please revert this space change

**Review by howard0su — APPROVED** [review]

File: ExtIO_sddc/resource.h [hunk 4]
```diff
@@ -77,7 +77,7 @@
 #define IDC_STATIC                      -1
 
 
-#ifndef _DEBUG  // define in processor setup resource !
+#ifdef NDEBUG  // define in processor setup resource !
```
Comment (howard0su) [inline ExtIO_sddc/resource.h:80]: maybe we define this out of resource.h.

I can see _DEBUG is defined. which may not apply to resource compile.
<PreprocessorDefinitions>TRACE;WIN32;_DEBUG;EXTIORX103_EXPORTS;_WINDOWS;_USRDLL;_CRT_SECURE_NO_WARNINGS;%(PreprocessorDefinitions)</PreprocessorDefinitions>

File: ExtIO_sddc/r2iq.cpp (hunk 1 above)
Comment (ik1xpv) [reply ExtIO_sddc/r2iq.cpp:78]: Yes. I'll do. Dynamic ADC frequency is a good feature !

**Comment by howard0su** [comment]
Looks like this make VHF only have 5M bandwidth on FM band. Is the issue you are working as well?

File: ExtIO_sddc/r2iq.cpp (hunk 3 above)
Comment (ik1xpv) [reply ExtIO_sddc/r2iq.cpp]: done

File: ExtIO_sddc/resource.h (hunk 4 above)
Comment (ik1xpv) [reply ExtIO_sddc/resource.h:80]: here my one 
TRACE;WIN32;_DEBUG;_WINDOWS;_USRDLL;_CRT_SECURE_NO_WARNINGS;%(PreprocessorDefinitions)
In italy seems not to work ' may be the "language" :-)) 
EXTIORX103_EXPORTS I think can be removed

**Comment by ik1xpv** [comment]
Yes. It seems to me a bug in low SR filtering.

File: ExtIO_sddc/r2iq.cpp [hunk 5]
```diff
@@ -373,41 +372,35 @@ void * r2iqControlClass::r2iqThreadf(r2iqThreadArg *th) {
...
-					}
-					else
-					{
-						th->inFreqTmp[m][0] = 0.0;
-						th->inFreqTmp[m][1] = 0.0;
-					}
-				}
+					int mm = _mtunebin + m;
+					if ((_mtunebin + m) < mfft /2)
+						mm -= - mfft /2;
```
Comment (howard0su) [inline ExtIO_sddc/r2iq.cpp:381]: mm += mfft / 2;
this will be more readable.

File: ExtIO_sddc/r2iq.cpp [hunk 6]
```diff
@@ -355,7 +354,7 @@ void * r2iqControlClass::r2iqThreadf(r2iqThreadArg *th) {
 		{
 			// FFT first stage time to frequency, real to complex
 			fftwf_execute_dft_r2c(th->plan_t2f_r2c, th->ADCinTime[k], th->ADCinFreq);
-
+			/*
```
Comment (howard0su) [inline ExtIO_sddc/r2iq.cpp:357]: You can delete the code. Git will keep all the historic code there.

**Review by howard0su — APPROVED** [review]

**Comment by ik1xpv** [comment]
Yes.

**Pull request merged**